utils.go

```diff
@@ -0,0 +1,51 @@
+package dpop
+
+import (
+	"crypto"
+	"encoding/base64"
+)
+
+// Structure which bundles a string to be hashed
+// with the associated hash function
+// Used in utils library to ensure hash function is available to user
+type HashInput struct {
+	InString string
+	HashFn	 crypto.Hash
+}
+
+// Internal function used to ensure hash is available and set the default
+// to SHA256
+func validateHashFunction(hashInput *HashInput) crypto.Hash {
+	if hashInput.HashFn.Available() {
+		return hashInput.HashFn
+	}
+	hashInput.HashFn = crypto.SHA256
+	return crypto.SHA256
```

Comment:
Either return a value or set it in the input. I think you should simply return it and not have a pointer in the function call.